odoo/tools/sql.py

```diff
@@ -152,6 +152,9 @@ def __bool__(self):
     def __eq__(self, other):
         return isinstance(other, SQL) and self.__code == other.__code and self.__params == other.__params
 
+    def __hash__(self):
+        return hash(self.__code, self.__params)
```

Comment:
Ok with the idea but hash only accept one argument :thinking: 
You can add at least one line of test for it.

Reply:
Test added.